src/track-type-dependency.js

```diff
@@ -9,8 +10,21 @@ export default function trackTypeDependency(typeName) {
   types[typeName] = true;
 }
 
+export function trackFieldDependency(typeName, fieldName) {
+  if (!tracking) {
+    return;
+  }
+
+  if (!fields[typeName]) {
+    fields[typeName] = {};
+  }
+
+  fields[typeName][fieldName] = true;
```

Comment:
Oh yeah 😶  I guess a `Set` could be used  but it would need a shim. A `reduce` is being used anyway though. I leave it up to you